ostelco-ios-client/ViewControllers/EKYC/SelectIdentityVerificationMethodViewController.swift

```diff
@@ -60,23 +61,37 @@ class SelectIdentityVerificationMethodViewController: UIViewController {
         showNeedHelpActionSheet()
     }
     
-    func getMyInfoURL() -> URL? {
-        var components = URLComponents(string: Environment().configuration(PlistKey.MyInfoURL))!
-        components.queryItems = [
-            URLQueryItem(name: "client_id", value: Environment().configuration(PlistKey.MyInfoClientID)),
-            //TODO: Find the right values for the query parameters.
-            //URLQueryItem(name: "attributes", value: "name,nationality,dob,email,mobileno,regadd"),
-            URLQueryItem(name: "attributes", value: "name,sex,race,nationality,dob,email,mobileno,regadd,housingtype,hdbtype,marital,edulevel,assessableincome,ownerprivate,assessyear,cpfcontributions,cpfbalances"),
-            URLQueryItem(name: "purpose", value: "eKYC"),
-            URLQueryItem(name: "state", value: "123"),
-            URLQueryItem(name: "redirect_uri", value: Environment().configuration(PlistKey.MyInfoCallbackURL)),
-        ]
-        return components.url
+    func startMyInfoLogin() {
+        self.spinnerView = self.showSpinner(onView: self.view)
+        // Fetch the configuration from prime
+        APIManager.shared.primeAPI
+            .loadMyInfoConfig()
+            .ensure { [weak self] in
+                self?.removeSpinner(self?.spinnerView)
+                self?.spinnerView = nil
+            }
+            .done { [weak self] myInfoConfig in
+                debugPrint("MyInfoConfig.url: \(myInfoConfig.url)")
+                var components = URLComponents(string: myInfoConfig.url)!
+                // Add purpose and state parameters to the url.
+                // state parameter is currently ignored by prime.
+                components.queryItems = [
+                    URLQueryItem(name: "purpose", value: "eKYC"),
+                    URLQueryItem(name: "state", value: "123")
```

Comment:
Why do we need the state parameter if it's ignored?

Reply:
It is mandatory  for MyInfo, we could use this in our backend for verification.  The uniqueness of this parameter is not imposed by MyInfo 
https://www.ndi-api.gov.sg/library/trusted-data/myinfo/tutorial2

Reply:
Prime is not using this now

Reply:
OK - can you please update the comment to include the bit about it being mandatory for myinfo?